src/utils/Fetcher.ts

```diff
@@ -1,96 +1,147 @@
-import ms from "ms";
-import AbortController from 'abort-controller';
+import ms from 'ms'
+import AbortController from 'abort-controller'
 import crossFetch from 'cross-fetch'
 import blobToBuffer from 'blob-to-buffer'
-import { clearTimeout, setTimeout } from "timers"
-import { applyDefaults, retry } from "./Helper";
+import { clearTimeout, setTimeout } from 'timers'
+import { retry } from './Helper'
 
 export class Fetcher {
+  private readonly defaults = {
+    method: 'GET',
+    responseType: ResponseType.JSON,
+    attempts: 1,
+    waitTime: '1s',
+    timeout: '5m'
+  }
 
-    constructor(private readonly defaultJsonRequestTimeout: string = '30s',
-        private readonly defaultFileDownloadRequestTimeout: string = '1m',
-        private readonly defaultPostTimeout: string = '5m') {}
-
-    async fetchJson(url: string, options?: RequestOptions): Promise<any> {
-        const opts = applyDefaults({ attempts: 1,
-            timeout: this.defaultJsonRequestTimeout,
-            waitTime: '0.5s',
-        }, options)
-        return this.fetchInternal(url, response => response.json(), opts)
+  private readonly fetchJsonDefaults = {
+    ...this.defaults,
+    ...{
+      timeout: '30s',
+      waitTime: '0.5s'
     }
-
-    async fetchBuffer(url: string, options?: RequestOptions): Promise<Buffer> {
-        const opts = applyDefaults({ attempts: 1,
-            timeout: this.defaultFileDownloadRequestTimeout,
-            waitTime: '1s',
-        }, options)
-        return this.fetchInternal(url, response => this.extractBuffer(response), opts)
+  }
+  private readonly fetchBufferDefaults = {
+    ...this.defaults,
+    ...{
+      timeout: '1m',
+      responseType: ResponseType.Buffer
     }
+  }
+  private readonly postDefaults = {
+    ...this.defaults,
+    ...{
+      method: 'POST'
+    }
+  }
 
-    async postForm(url: string, body: FormData | string, headers: any = { }, options?: RequestOptions): Promise<any> {
-        const opts = applyDefaults({ attempts: 1,
-            timeout: this.defaultPostTimeout,
-            waitTime: '1s',
-        }, options)
+  constructor() {}
```

Comment:
I think it would be nice to be able to configure the defaults when you create the fetcher. That way, you can have the same custom defaults cross-project.

We are already doing that on the content server, with timeouts. And it would be nice to do the same with the user-agent header. Just set it when the `Fetcher` is created, and forget about it. With the current change, we would need to pass the user-agent header each time a `fetch` is called

Reply:
Done!